python/src/energuide/dwelling.py

```diff
@@ -89,25 +140,46 @@ class ParsedDwellingDataRow(_ParsedDwellingDataRow):
         'CLIENTCITY': {'type': 'string', 'required': True},
         'forwardSortationArea': {'type': 'string', 'required': True, 'regex': '[A-Z][0-9][A-Z]'},
         'HOUSEREGION': {'type': 'string', 'required': True},
+
+        'ceilings': {
+            'type': 'list',
+            'required': True,
+            'schema': {
+                'type': 'dict',
+                'schema': {
+                    'label': {'type': 'string', 'required': True},
+                    'type_english': {'type': 'string', 'required': True},
+                    'type_french': {'type': 'string', 'required': True},
+                    'nominal_rsi': {'type': 'float', 'required': True, 'coerce': float},
+                    'effective_rsi': {'type': 'float', 'required': True, 'coerce': float},
+                    'area': {'type': 'float', 'required': True, 'coerce': float},
+                    'length': {'type': 'float', 'required': True, 'coerce': float}
+                }
+            }
+        }
     }
 
     @classmethod
     def from_row(cls, row: reader.InputData) -> 'ParsedDwellingDataRow':
         validator = cerberus.Validator(cls._SCHEMA, allow_unknown=True)
         if not validator.validate(row):
+            # import pdb; pdb.set_trace()
             error_keys = ', '.join(validator.errors.keys())
             raise reader.InvalidInputDataException(f'Validator failed on keys: {error_keys}')
 
+        parsed = validator.document
+
         return ParsedDwellingDataRow(
-            eval_id=row['EVAL_ID'],
-            eval_type=EvaluationType.from_code(row['EVAL_TYPE']),
-            entry_date=parser.parse(row['ENTRYDATE']).date(),
-            creation_date=parser.parse(row['CREATIONDATE']),
-            modification_date=parser.parse(row['MODIFICATIONDATE']),
-            year_built=row['YEARBUILT'],
-            city=row['CLIENTCITY'],
-            region=Region.from_data(row['HOUSEREGION']),
-            forward_sortation_area=row['forwardSortationArea']
+            eval_id=parsed['EVAL_ID'],
+            eval_type=EvaluationType.from_code(parsed['EVAL_TYPE']),
+            entry_date=parser.parse(parsed['ENTRYDATE']).date(),
+            creation_date=parser.parse(parsed['CREATIONDATE']),
+            modification_date=parser.parse(parsed['MODIFICATIONDATE']),
```

Comment:
Should we put `parser.parse` in the `coerce` method of the validator?

Reply:
Not unless we want to make a custom validator type. The coercion is applied before the validation, so if it coerced it into a type that isn't supported by default, we'd have to write a custom validation type for it, except if it didn't throw an exception when coercing, we automatically know it's valid, so it's a weird situation